bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/Storage.java

```diff
@@ -1778,37 +1762,51 @@ private Properties findVMProfile(Generation systemGeneration) {
 		return result;
 	}
 
+	private InputStream createProfileStream(Generation systemGeneration, String vmProfile,
+			String embeddedProfileName) {
+		InputStream profileIn = null;
+		if (vmProfile != null) {
+			// look for a profile in the system bundle based on the vm profile
+			String javaProfile = vmProfile + PROFILE_EXT;
+			profileIn = findInSystemBundle(systemGeneration, javaProfile);
+			if (profileIn == null)
+				profileIn = getNextBestProfile(systemGeneration, JAVASE, runtimeVersion, embeddedProfileName);
+		}
+		if (profileIn == null)
+			// the profile url is still null then use the min profile the framework can use
+			profileIn = findInSystemBundle(systemGeneration, "JavaSE-1.8.profile"); //$NON-NLS-1$
+		return profileIn;
+	}
+
 	private Properties readConfiguredJavaProfile(Generation systemGeneration) {
 		// check for the java profile property for a url
 		String propJavaProfile = equinoxContainer.getConfiguration().getConfiguration(EquinoxConfiguration.PROP_OSGI_JAVA_PROFILE);
 		if (propJavaProfile != null) {
-			InputStream profileIn = null;
-			try {
-				// we assume a URL
-				profileIn = new URL(propJavaProfile).openStream();
-			} catch (IOException e) {
-				// try using a relative path in the system bundle
-				profileIn = findInSystemBundle(systemGeneration, propJavaProfile);
-			}
-			if (profileIn != null) {
-				Properties result = new Properties();
-				try {
-					result.load(new BufferedInputStream(profileIn));
-				} catch (IOException e) {
-					// consider logging
-				} finally {
-					try {
-						profileIn.close();
-					} catch (IOException e) {
-						// nothing to do
+			try (InputStream profileIn = createPropStream(systemGeneration, propJavaProfile)){
+				if (profileIn != null) {
+					Properties result = new Properties();
+					try (BufferedInputStream inStream = new BufferedInputStream(profileIn)) {
```

Comment:
Same as above, another buffer is probably not necessary and `result.load(profileIn);` would probably be enough and even better.